pkg/network/dns/driver_windows.go

```diff
@@ -61,18 +61,38 @@ func (d *dnsDriver) setupDNSHandle() error {
 		return err
 	}
 
-	if err := dh.SetDataFilters(filters); err != nil {
+	// must assign handle value before calling SetDataFilters because SetDataFilters
+	// uses it.
+	d.h = dh
+	if err := d.SetDataFilters(filters); err != nil {
 		return err
 	}
 
-	iocp, buffers, err := prepareCompletionBuffers(dh.Handle, dnsReadBufferCount)
+	iocp, buffers, err := prepareCompletionBuffers(dh.GetWindowsHandle(), dnsReadBufferCount)
```

Comment:
You likely should move `prepareCompletionBuffer` to the `Handle` interface. That would allow you to get rid of `GetWindowsHandle` which kind of breaks your nice `Handle` interface abstraction.

Reply:
Thank you for the suggestion. As discussed in Slack, will move to different PR as needed.